implants/lib/eldritch/src/pivot/ssh_exec_impl.rs

```diff
@@ -1,5 +1,205 @@
 use anyhow::Result;
+use starlark::{values::{dict::Dict, Heap, Value}, collections::SmallMap, const_frozen_string};
 
-pub fn ssh_exec(_target: String, _port: i32, _username: String, _password: String, _key: String, _command: String, _shell_path: String) -> Result<String> {
-    unimplemented!("Method unimplemented")
+use super::Session;
+
+struct SSHExecOutput {
+    stdout: String,
+    status: i32,
+}
+
+async fn handle_ssh_exec(target: String, port: u16, command: String, username: String, password: Option<String>, key: Option<String>, key_password: Option<&str>, timeout: Option<u32>) -> Result<SSHExecOutput> {
+    let mut ssh = tokio::time::timeout(
+        std::time::Duration::from_secs(timeout.unwrap_or(3).into()),
+        Session::connect(username, password, key, key_password, format!("{}:{}", target, port)),
+
+    ).await??;
+    let r = ssh.call(&command).await?;
+    ssh.close().await?;
+
+    Ok(
+        SSHExecOutput { 
+            stdout: r.output(),
+            status: r.code.unwrap_or(0) as i32,
+        }
+    )
+}
+
+pub fn ssh_exec(starlark_heap: &Heap, target: String, port: i32, command: String, username: String, password: Option<String>, key: Option<String>, key_password: Option<String>, timeout: Option<u32>) -> Result<Dict> {
+    
+    let runtime = tokio::runtime::Builder::new_current_thread()
+        .enable_all()
+        .build()
+        .unwrap();
+
+    let key_password_ref = key_password.as_deref();
+    let local_port: u16 = port.try_into()?;
+
+    let cmd_res = match runtime.block_on(
+        handle_ssh_exec(target, local_port, command, username, password, key, key_password_ref, timeout)
+    ) {
+        Ok(local_res) => local_res,
+        Err(local_err) => return Err(anyhow::anyhow!("Failed to run handle_ssh_exec: {}", local_err.to_string())),
+    };
+
+    let res = SmallMap::new();
+    let mut dict_res = Dict::new(res);
+    let stdout_value = starlark_heap.alloc_str(&cmd_res.stdout);
+    dict_res.insert_hashed(const_frozen_string!("stdout").to_value().get_hashed().unwrap(), stdout_value.to_value());
+
+    let status_value = Value::new_int(cmd_res.status);
+    dict_res.insert_hashed(const_frozen_string!("status").to_value().get_hashed().unwrap(), status_value);
+
+    Ok(dict_res)
+}
+
+#[cfg(test)]
+mod tests {
+    use async_trait::async_trait;
+    use tokio::net::TcpListener;
+    use tokio::task;
+    use std::process::Command;
+    use std::sync::{Mutex, Arc};
+    use russh::*;
+    use russh::server::{Auth, Session, Msg};
+    use russh_keys::*;
+    use std::collections::HashMap;
+    use super::*;
+
+    // SSH Server utils
+    #[derive(Clone)]
+    #[allow(dead_code)]
```

Comment:
the sussiest baka

Reply:
why dis here?

Reply:
The ssh library boiler plate has a bunch of sections that while actually used lint as dead_code. Here `client_pubkey` gets linted as unused.